Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.java

```diff
@@ -0,0 +1,173 @@
+package com.automattic.simplenote;
+
+import android.os.Bundle;
+import android.os.Handler;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.text.method.LinkMovementMethod;
+import android.util.TypedValue;
+import android.view.ContextThemeWrapper;
+import android.view.View;
+import android.widget.Button;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.automattic.simplenote.models.Tag;
+import com.automattic.simplenote.utils.DisplayUtils;
+import com.automattic.simplenote.utils.HtmlCompat;
+import com.automattic.simplenote.utils.TagUtils;
+import com.automattic.simplenote.utils.ThemeUtils;
+import com.automattic.simplenote.widgets.MorphCircleToRectangle;
+import com.automattic.simplenote.widgets.MorphSetup;
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+import com.simperium.client.Bucket;
+import com.simperium.client.BucketObjectNameInvalid;
+
+import java.util.Objects;
+
+public class AddTagActivity extends AppCompatActivity implements TextWatcher {
+    private Bucket<Tag> mBucketTag;
+    private Button mButtonNegative;
+    private Button mButtonPositive;
+    private TextInputEditText mTagInput;
+    private TextInputLayout mTagLayout;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        ThemeUtils.setTheme(this);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_tag_add);
+
+        int widthScreen = getResources().getDisplayMetrics().widthPixels;
+        int widthMargin = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 56, getResources().getDisplayMetrics());
+        int widthMaximum = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 312, getResources().getDisplayMetrics());
+        LinearLayout layout = findViewById(R.id.layout);
+        layout.getLayoutParams().width = Math.min(widthMaximum, widthScreen - widthMargin);
+        layout.requestLayout();
```

Comment:
just a np, why ConstraintLayouts are not used in the project? if they were used, you can have this logic in the XML layout, and would be easier to understand.

Reply:
`ConstraintLayout`s were never required to do what we need to with Simplenote.  So, they weren't added.  It's debatable which view layout is more understandable when you've been working on Android before the time of `ConstraintLayout` and you're used to `LinearLayout` and `RelativeLayout` for everything.